Tests/Marketplace/marketplace_services.py

```diff
@@ -3007,7 +3026,212 @@ def copy_author_image(self, production_bucket, build_bucket, images_data, storag
+                detect_changes (bool): Whether to detect changes or upload the author image in any case.
+                marketplace (str): The marketplace the pack is linked to.
+            Returns:
+                bool: whether the operation succeeded.
+        """
+        task_status = True
+
+        try:
+            pack_readme_path = os.path.join(self._pack_path, Pack.README)  # disable-secrets-detection
+
+            if not os.path.exists(pack_readme_path):
+                return True
+
+            storage_pack_path = os.path.join(storage_base_path, self._pack_name)  # disable-secrets-detection
+
+            if not detect_changes or any(self.is_raedme_file(file.a_path) for file in diff_files_list):
+                # detect added/modified integration readme files
+                logging.info(f'found a pack: {self._pack_name} with changes in README')
+                readme_images_storage_paths = self.collect_images_from_readme_and_replace_with_storage_path(
+                    pack_readme_path, storage_pack_path, marketplace)
+
+                # no external image urls were found in the readme file
+                if not readme_images_storage_paths:
+                    logging.info(f'no image links were found in {self._pack_name} readme file')
+                    return task_status
+
+                for image_info in readme_images_storage_paths:
+                    readme_original_url = image_info.get('original_read_me_url')
+                    gcs_storage_path = str(image_info.get('new_gcs_image_path'))
+                    image_name = str(image_info.get('image_name'))
+
+                    task_status = self.download_readme_image_from_url_and_upload_to_gcs(readme_original_url,
+                                                                                        gcs_storage_path,
+                                                                                        image_name, storage_bucket)
+                    self._reademe_images.append(image_name)
+
+        except Exception:
+            logging.exception(f"Failed uploading {self._pack_name} pack readme image.")
+            task_status = False
+        finally:
+            return task_status
+
+    def collect_images_from_readme_and_replace_with_storage_path(self, pack_readme_path, gcs_pack_path, marketplace):
+        """
+        Replaces inplace all images links in the pack README.md with their new gcs location
+
+        Args:
+            pack_readme_path (str): A path to the pack README file.
+            gcs_pack_path (str): A path to the pack in gcs
+            marketplace (str): The marketplace this pack is going to be uploaded to.
+
+        Returns:
+            A list of dicts of all the image urls found in the README.md file with all related data
+            (original_url, new_gcs_path, image_name)
+        """
+        logging.info(f'{marketplace=}')
+        if marketplace == 'xsoar':
+            marketplace = 'marketplace-dist'
```

Comment:
I suggest to name it:
```suggestion
            marketplace_bucket = 'marketplace-dist'
```
so it will not confuse with the original marketplace argument.

Reply:
Done